doc/packed_data.rdoc

```diff
@@ -0,0 +1,586 @@
+    # => [67305985, 4244504319]
+
+==== Pointer Directives
+
+- <tt>'j'</tt> - 64-bit pointer-width signed integer,
+  native-endian (like C <tt>intptr_t</tt>):
+
+    s = [67305985, -50462977].pack('j*')
+    # => "\x01\x02\x03\x04\x00\x00\x00\x00\xFF\xFE\xFD\xFC\xFF\xFF\xFF\xFF"
+    s.unpack('j*')
+    # => [67305985, -50462977]
+
+- <tt>'j'</tt> - 64-bit pointer-width unsigned integer,
+  native-endian (like C <tt>uintptr_t</tt>):
+
+    s = [67305985, 4244504319].pack('J*')
+    # => "\x01\x02\x03\x04\x00\x00\x00\x00\xFF\xFE\xFD\xFC\x00\x00\x00\x00"
+    s.unpack('J*')
+    # => [67305985, 4244504319]
+
+==== Other \Integer Directives
+:
+- <tt>'U'</tt> - UTF-8 character:
+
+    s = [4194304].pack('U*')
+    # => "\xF8\x90\x80\x80\x80"
+    s.unpack('U*')
+    # => [4194304]
+
+- <tt>'w'</tt> - BER-encoded integer
+  (see {BER enocding}[https://en.wikipedia.org/wiki/X.690#BER_encoding]):
+
+    s = [1073741823].pack('w*')
+    # => "\x83\xFF\xFF\xFF\x7F"
+    s.unpack('w*')
+    # => [1073741823]
+
+==== Modifiers for \Integer Directives
+
+For directives in
+<tt>'i'</tt>,
+<tt>'I'</tt>,
+<tt>'s'</tt>,
+<tt>'S'</tt>,
+<tt>'l'</tt>,
+<tt>'L'</tt>,
+<tt>'q'</tt>,
+<tt>'Q'</tt>,
+<tt>'j'</tt>, and
+<tt>'J'</tt>,
+these modifiers may be suffixed:
+
+- <tt>'!'</tt> or <tt>'_'</tt> - Underlying platform’s native size.
+- <tt>'>'</tt> - Big-endian.
+- <tt>'<'</tt> - Little-endian.
+
+=== \Float Directives
+
+Each float directive specifies the packing or unpacking
+for one element in the input or output array.
+
+==== Single-Precision \Float Directives
+
+- <tt>'F'</tt> or <tt>'f'</tt> - Native format:
+
+    s = [3.0].pack('F') # => "\x00\x00@@"
+    s.unpack('F')       # => [3.0]
+
+- <tt>'e'</tt> - Little-endian:
+
+    s = [3.0].pack('e') # => "\x00\x00@@"
+    s.unpack('e')       # => [3.0]
+
+- <tt>'g'</tt> - Big-endian:
+
+    s = [3.0].pack('g') # => "@@\x00\x00"
+    s.unpack('g')       # => [3.0]
+
+==== Double-Precision \Float Directives
+
+- <tt>'D'</tt> or <tt>'d'</tt> - Native format:
+
+    s = [3.0].pack('D') # => "\x00\x00\x00\x00\x00\x00\b@"
+    s.unpack('D')       # => [3.0]
+
+- <tt>'E'</tt> - Little-endian:
+
+    s = [3.0].pack('E') # => "\x00\x00\x00\x00\x00\x00\b@"
+    s.unpack('E')       # => [3.0]
+
+- <tt>'G'</tt> - Big-endian:
+
+    s = [3.0].pack('G') # => "@\b\x00\x00\x00\x00\x00\x00"
+    s.unpack('G')       # => [3.0]
+
+A float directive may be infinity or not-a-number:
+
+  inf = 1.0/0.0                  # => Infinity
+  [inf].pack('f')                # => "\x00\x00\x80\x7F"
+  "\x00\x00\x80\x7F".unpack('f') # => [Infinity]
+
+  nan = inf/inf                  # => NaN
+  [nan].pack('f')                # => "\x00\x00\xC0\x7F"
+  "\x00\x00\xC0\x7F".unpack('f') # => [NaN]
+
+=== \String Directives
+
+Each string directive specifies the packing or unpacking
+for one byte in the input or output string.
+
+==== Binary \String Directives
+
+- <tt>'A'</tt> - Arbitrary binary string (space padded; count is width)
```

Comment:
Should there be a period at the end of this line? The next line feels like a new sentence to me.

Reply:
Fixed.